xcode/Subconscious/Shared/Components/Common/Forms/ValidatedTextField.swift

```diff
@@ -6,39 +6,47 @@
 //
 
 import SwiftUI
+import Combine
 
 /// A text field that comes with help text and a validation flag
 struct ValidatedTextField: View {
     var placeholder: String
     @Binding var text: String
+    var onFocusChanged: ((Bool) -> Void)?
```

Comment:
Comment: I typically prefer to require a closure so I don't have to use unwrapping syntax. It's easy to provide an empty closure for no-op.

No change necessary.